src/dash-playlist-loader.js

```diff
@@ -277,20 +282,15 @@ export default class DashPlaylistLoader extends EventTarget {
 
     // initialize the loader state
     // The masterPlaylistLoader will be created with a string
-    if (typeof srcUrlOrPlaylist === 'string') {
+    if (this.isMaster_) {
       this.srcUrl = srcUrlOrPlaylist;
       // TODO: reset sidxMapping between period changes
       // once multi-period is refactored
-      this.sidxMapping_ = {};
-      return;
+      this.masterPlaylistLoader_.sidxMapping_ = {};
```

Comment:
Since we have the conditional above for master loaders, would it be clearer to use just `this`?